File: Juan/P2/java/Napakalaki/src/napakalaki/PruebaNapakalaki.java
package napakalaki;

import java.util.ArrayList;
import java.util.Arrays;

/**
 *
 * @author juane
 * @version 2.1
 */
public class PruebaNapakalaki {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
     
       BadConsequence badConsequence ;  // mal royo
       Prize prize; // buen royo
      
       ArrayList<Monster> monstruos = new ArrayList(); // Lista de Monstrous
       ArrayList<TreasureKind> tVisible = new ArrayList(); // Lista de TreasureKind Visible
       ArrayList<TreasureKind> tHidden = new ArrayList(); // Lista de TresaureKind Oculto
       
       // 3 Byakhees de bonanza
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.ARMOR));
       tHidden = new ArrayList(Arrays.asList(TreasureKind.ARMOR));
       badConsequence = new BadConsequence("Pierdes tu armadura visible y otra oculta"
               + "",0,tVisible,tHidden);
       prize = new Prize(2,1);
       monstruos.add(new Monster("3 Byakhees de bonanza ",8,badConsequence,prize));
       
       // Chibithulhu 
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.HELMET));
       tHidden = new ArrayList();
       badConsequence = new BadConsequence("Embobados con el lindo primigenio te descartas de tu"
               + "casco visible",0,tVisible,tHidden);
       prize = new Prize(1,1);
       monstruos.add(new Monster("Chibithulhu",2,badConsequence,prize));
       
       // El sopor de Dunwich
       tVisible = new ArrayList(Arrays.asList(TreasureKind.SHOES));
       tHidden = new ArrayList();
       badConsequence = new BadConsequence("El primordial bostezo contagioso.Pierdes el calzado visible"
                ,0,tVisible,tHidden);
       prize = new Prize(1,1);
       monstruos.add(new Monster("El sopor de Dunwich",2,badConsequence,prize));       
       
       
        // Ángeles de la noche ibicenca
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.ONEHAND));
       tHidden = new ArrayList(Arrays.asList(TreasureKind.ONEHAND));
       badConsequence = new BadConsequence("Te atrapan para llevarte de fiesta "
               + "y te dejan caer en mitadl del vuelo. Descarta 1 mano visible "
               + "y 1 mano oculta",0,tVisible,tHidden);
       prize = new Prize(4,1);
       monstruos.add(new Monster("Ángeles de la noche ibicenca",14,badConsequence,prize));
       
       //El gorrón en el umbral  
       
       badConsequence = new BadConsequence("Pierdes todos tus tesoros visibles",0,10,0);
       prize = new Prize(3,1);       
       monstruos.add(new Monster("El gorrón en el umbral",10,badConsequence,prize));
       
       // H.P. Munchcraft
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.ARMOR));
       tHidden = new ArrayList();
       badConsequence = new BadConsequence("Pierdes la armadura visible",0,tVisible,tHidden);
       prize = new Prize(2,1);
       monstruos.add(new Monster("H.P Munchcraft",6,badConsequence,prize));
         
       
       // Bichgooth
       tVisible = new ArrayList(Arrays.asList(TreasureKind.ARMOR));
       tHidden = new ArrayList();
       badConsequence = new BadConsequence("Sientes bichos bajo la ropa.Descarta la armadura visible"
               ,0,tVisible,tHidden);
       prize = new Prize(1,1);
       monstruos.add(new Monster("Bichgooth",2,badConsequence,prize));
       
       
        
       // El rey de rosa
       badConsequence = new BadConsequence("Pierdes 5 niveles y 3 tesoros visibles",5,3,0);
       prize = new Prize(4,2);
       monstruos.add(new Monster("El rey de rosa",13,badConsequence,prize));
       
        //La que redacta en las tinieblas
       
       badConsequence = new BadConsequence("Toses los pulmones y "
               + "pierdes 2 niveles",2,0,0);
       prize = new Prize(1,1);
       monstruos.add(new Monster("La que redacta en las tinieblas",2
               ,badConsequence,prize));
       
       
       // Los hondos
       
       badConsequence = new BadConsequence("Estos monstrous resultan"
               + "bastante superficiales y te aburren mortalmente."
               + "Estas muerto",true);
       prize = new Prize(2,1);
       monstruos.add(new Monster("Los hondos",8,badConsequence,prize));
       
       
       // Semillas Cthulhu
       
       badConsequence = new BadConsequence("Pierdes 2 niveles y 2 tesoros "
               + "ocultos",2,0,2);
       prize = new Prize(2,1);
       monstruos.add(new Monster("Semillas Cthulhu",4,badConsequence,prize));
       
       // Dameargo
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.ONEHAND));
       tHidden = new ArrayList();
       prize = new Prize(2,1);
       badConsequence = new BadConsequence("Te intentas escaquear. Pierdes"
               + "una mano visible",0,tVisible,tHidden);
       
       monstruos.add(new Monster("Dameargo",1,badConsequence,prize));
       
       // Pollipólipo volante
       
       badConsequence = new BadConsequence("Da mucho asquito. Pierdes"
               + "3 niveles",3,0,0);
       prize = new Prize(1,1);
       monstruos.add(new Monster("Pollipólipo volante",3,badConsequence,prize));
       
       // Yskhtihyssg-Goth
       
       badConsequence = new BadConsequence("No le hace gracia que pronuncien mal "
               + "su nombre. Estas muerto",true);
       prize = new Prize(3,1);
       monstruos.add(new Monster("Yskhtihyssg-Goth",12,badConsequence,prize));
       
       // Familia feliz
       
       badConsequence = new BadConsequence("La familia te atrapa.Estás muerto"
               + "",true);
       prize = new Prize(4,1);
       monstruos.add(new Monster("Familia Feliz",1,badConsequence,prize));
       
       
       //Roboggoth
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.BOTHHANDS));
       tHidden = new ArrayList();
       prize = new Prize(2,1);
       badConsequence = new BadConsequence("La quinta directiva primaria te obliga"
               + " a perder 2 niveles y un tesoro, 2 manos visibles",2,tVisible,tHidden);
       monstruos.add(new Monster("Roboggoth",8,badConsequence,prize));
       
       //El espia
       
       tVisible = new ArrayList(Arrays.asList(TreasureKind.HELMET));
       tHidden = new ArrayList();
       badConsequence = new BadConsequence("Te asusta en la noche. Pierdes un "
               + "casco visible",0,tVisible,tHidden);
       prize = new Prize(1,1);
       monstruos.add(new Monster("El espia",5,badConsequence,prize));
       
       // El Lenguas
       
       badConsequence = new BadConsequence("Menudo susto te llevas."
               + " Pierdes 2 niveles y 5 tesoros visibles",2,5,0);
       prize = new Prize(1,1);
       monstruos.add(new Monster("El Lenguas",20,badConsequence,prize));
       
       // Bicéfalo
       tVisible = new ArrayList(Arrays.asList(TreasureKind.ONEHAND));
       tHidden = new ArrayList(Arrays.asList(TreasureKind.BOTHHANDS));;
       badConsequence = new BadConsequence("Te faltan manos para tanta cabeza. "
               + "Pierdes 3 niveles y tus tesoros visibles de las manos",3,tVisible,tHidden);
       prize = new Prize(1,1);
       monstruos.add(new Monster("Bicéfalo",20,badConsequence,prize));
       
       
       /*Consultas*/
       Monster mon;
       int total = 0;
       
       System.out.println("Nivel de Combate superior a 10"+"\n");
        for (int i = 0; i < monstruos.size(); i++) {
            mon = monstruos.get(i);
            
            if (mon.getCombatLevel() > 10) {
                total++;

                System.out.println(monstruos.get(i) + "\n");
            }
        }
       
       System.out.println("Total:" + total+"\n");
       
       
       
       int perLevel;
       
       total = 0;
       System.out.println("Mal rollo que implica sólo la pérdida de niveles"+"\n");
       
       for(int i = 0; i < monstruos.size();i++){
           perLevel = monstruos.get(i).getBadConsequence().getLevels();
         
           if(perLevel > 0 ){
               System.out.println(monstruos.get(i)+"\n");
               total++;
           }
           
          
       }
       System.out.println("Total: " + total);
       
       
       total = 0;
       int level_prize;
       System.out.println("buen rollo ganancia de niveles superor a 1"+"\n");
       
       for(int i = 0; i < monstruos.size();i++){
           level_prize = monstruos.get(i).getPrice().getLevel();
           
           if(level_prize > 1){
               System.out.println(monstruos.get(i)+"\n");
               total++;
           }
       }
       System.out.println("Total: " + total);
       
       total = 0;
      
       for(int i = 0; i < monstruos.size();i++){
            tVisible = monstruos.get(i).getBadConsequence().getSpecificVisibleTreasures();
            tHidden = monstruos.get(i).getBadConsequence().getSpecificHiddenTreasures();
       
            if(!tVisible.isEmpty() || !tHidden.isEmpty()){
                System.out.println(monstruos.get(i)+"\n");               
                total++;
            }
       }
       
       System.out.println("Total " + total +"\n");
    }
    
}

File: Juan/P3/Ruby/Napakalaki/lib/PruebaNapakalaki.rb
#encoding: utf-8

#Versión 3.0

require_relative 'BadConsequence.rb'
require_relative 'Monster.rb'
require_relative 'TreasureKind.rb'
require_relative 'Prize.rb'
require_relative 'Napakalaki.rb'
require_relative 'CombatResult.rb'
#monsters = Array.new
# PRÁCTICA 1
## Monstrous
#
#
## 3 Byakhees de bonanza
#
#price = Prize.new(2,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Pierdes tu armadura visible y otra oculta',0,[TreasureKind::ARMOR],[TreasureKind::ARMOR])
#monsters << Monster.new('3 Byakhees de bonanza',8,badconsequence,price)
#  
## Chibithulhu
#  
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Embobados don el lindo primigenio te descartas de tu casco visible',0,[TreasureKind::HELMET],[])
#monsters << Monster.new('Chibithulhu',2,badconsequence,price)
#
## El sopor de Dunwich
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('El primordial bostezo contagioso. Pierdes el calzado visible',0,[TreasureKind::SHOES],[])
#monsters << Monster.new('El sopor de Dunwich',2,badconsequence,price)
#
#
## Ángeles de la noche ibicenca
#
#price = Prize.new(4,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y  1 mano oculta',0 ,[TreasureKind::ONEHAND],[TreasureKind::ONEHAND])
#monsters << Monster.new('Angeles de la noche ibicenca',14,badconsequence,price)
#
## El gorrón en el umbral
#
#price = Prize.new(3,1)
#badconsequence = BadConsequence.newLevelNumberOfTreasures('El gorron en el umbral',0,10,0)
#monsters << Monster.new('El gorron en el umbra',10,badconsequence,price)
#
## H.P. Munchcraft
#
#price = Prize.new(2,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Pierdes la armadura visible',0,[TreasureKind::ARMOR],[])
#monsters << Monster.new('H.P. Munchcraft',6,badconsequence,price)
#
## Bichgooth
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Sientes bichos bajo la ropa. Descarta la armadura visible',0,[TreasureKind::ARMOR] , [] )
#monsters << Monster.new('Bichgooth',2,badconsequence,price)
#
#
## El rey de rosa
#
#price = Prize.new(4,2)
#badconsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes 5 niveles y 3 tesoros visibles', 5, 3, 0)
#monsters << Monster.new('El rey de rosa',13,badconsequence,price)
#
## La que redacta en las tinieblas
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelNumberOfTreasures('Toses los pulmones y pierdes 2 niveles',2,0,0)
#monsters << Monster.new('La que redacta en las tinieblas',2,badconsequence,price)
#
## Los hondos
#
#price = Prize.new(2,1)
#badconsequence = BadConsequence.newDeath("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto")
#monsters << Monster.new('Los hondos',8,badconsequence,price)
#
## Semillas Cthulhu
#
#price = Prize.new(2,1)
#badconsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes 2 niveles y 2 tesoros ocultos',2,0,2)
#monsters << Monster.new('Semillas Cthulhu',4,badconsequence,price)
#
## Dameargo
#
#price = Prize.new(2,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Te intentas escaquear. Pierdes una mano visible',0,[TreasureKind::ONEHAND],[])
#monsters << Monster.new('Dameargo',1,badconsequence,price)
#
## Pollipólipo volante
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelNumberOfTreasures('Da mucho asquito. Pierdes 3 niveles.',3,0,0)
#monsters << Monster.new('Pollipolipo volante',3,badconsequence,price)
#
## Yskhtihyssg-Goth
#
#price = Prize.new(3,1)
#badconsequence = BadConsequence.newDeath('No le hace gracia que pronuncien mal su nombre. Estas muerto')
#monsters << Monster.new('Yskhtihyssg-Goth',12,badconsequence,price)
#
## Familia feliz
#
#price = Prize.new(4,1)
#badconsequence = BadConsequence.newDeath("La familia te atrapa. Estas muerto")
#monsters << Monster.new('Familia feliz',1,badconsequence, price)
#
## Roboggoth
#price = Prize.new(2,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('La quinta directeiva primaria te obliga a perder 2 niveles y un tesoro, 2 manos visibles.',2,[TreasureKind::BOTHHANDS],[])
#monsters << Monster.new('Roboggoth',8,badconsequence,price)
#
## El espía
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Te asusta en la noche. Pierdes un casco visible',0,[TreasureKind::HELMET],[])
#monsters << Monster.new('El espia',5,badconsequence,price)
#
## El Lenguas
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelNumberOfTreasures('Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles',2, 5,0)
#monsters << Monster.new('El Lenguas',20,badconsequence,price)
#
## Bicéfalo
#
#price = Prize.new(1,1)
#badconsequence = BadConsequence.newLevelSpecificTreasures('Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos',3,[TreasureKind::ONEHAND,TreasureKind::ONEHAND,TreasureKind::BOTHHANDS],[])
#monsters << Monster.new('Bicefalo',20,badconsequence,price)
#
##Bucle para imprimir el array de monstruos
#i = 0
##while i < monsters.size
##    puts monsters.fetch(i)
##    i = i+1
##end
#
##Mostrar todos los monstruos que tengan un nivel de combate superior a 10.
#num = 0
#puts "Todos los monstruos que tenga un vivel de combate superior a 10"+"\n\n"
#while i < monsters.size    
#    if (monsters.fetch(i).combatLevel > 10)
#        puts monsters.fetch(i)
#       num = num+1
#    end 
#    i = i+1
#end
#puts "\n"
## Mostrar todos los monstruos que tengan un mal rollo que implique sólo pérdida de niveles.
#
#i=0
#num =0
#puts "Todos los monstruos que tengan un mal rollo que implique solo perdida de niveles"+"\n\n"
#while i < monsters.size
#    
#    auxiliar = monsters[i].badconsequence.levels
#    
#    if(auxiliar > 0)
#        puts monsters.fetch(i)
#        num = num + 1
#    end
#    i = i+1
#end
#puts "\n"
## Mostrar todos los monstruos que tengan un buen rollo que indique una ganancia de
## niveles superior a 1.
#
#i=0
#num =0
#puts "Todos los monstruos que tengan un buen rollo que indique una gananacia de niveles superior a 1"+"\n\n"
#
#while i < monsters.size
#    auxiliar = monsters.fetch(i).price.levels #método get para acceder al elemento en la posición i
#    
#        if (auxiliar > 1)
#            puts monsters.fetch(i) 
#           num = num+1
#        end
#    i = i+1
#end
#puts "\n"
##Mostrar todos los monstruos que tengan un mal rollo que suponga la pérdida de un
##determinado tipo de tesoros ya sea visibles y/o ocultos. Debe mostrarse el
##nombre, nivel de combate, buen rollo y mal rollo de cada monstruo.
#i=0
#num = 0
#puts "Todos los monstruos que tengan un mal rollo que suponga la pérdida de un determinado tipo de tesoros ya sea visible y/o ocultos"+"\n\n"
#while i < monsters.size
#    specificV = monsters.fetch(i).badconsequence.specificVisibleTreasures
#    specificH = monsters.fetch(i).badconsequence.specificHiddenTreasures
#    
#    if (specificV != [] || specificH != [])
#        puts monsters.fetch(i)
#        num = num+1
#    end
#    i = i+1
#end
#puts "\n"
#
##Imprimimos el número de monstrous
#puts "Total monstrous: #{num}"


napa = Napakalaki.instance
res = [CombatResult::LOSE]
#res = [CombatResult::WINGAME]
puts napa.endOfGame(res)
